plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/helpers.ts

```diff
@@ -313,6 +313,115 @@ export async function initRepoPushAndProtect(
   }
 }
 
+export async function createGithubTemplateRepoWithCollaboratorsAndTopics(
+  client: Octokit,
+  repo: string,
+  owner: string,
+  repoVisibility: 'private' | 'public',
+  description: string | undefined,
+  templateOwner: string,
+  templateRepo: string,
+  access: string | undefined,
+  collaborators:
+    | (
+        | {
+            user: string;
+            access: 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
+          }
+        | {
+            team: string;
+            access: 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
+          }
+        | {
+            /** @deprecated This field is deprecated in favor of team */
+            username: string;
+            access: 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
+          }
+      )[]
+    | undefined,
+  logger: Logger,
+) {
+  // eslint-disable-next-line testing-library/no-await-sync-query
+  const user = await client.rest.users.getByUsername({
+    username: owner,
+  });
+
+  const templateRepoCreationPromise = client.rest.repos.createUsingTemplate({
```

Comment:
Before creating the template, we probably want to know if the given template url actually exists and if it's of type ``template`` right?
A check like this would solve it:
```suggestion
  const repoTemplatePromise = client.rest.repos.get({
        owner: templateOwner,
        repo: templateRepo,
      });

      let repoTemplate;
      try {
        repoTemplate = (await repoTemplatePromise).data;
      } catch (e) {
        assertError(e);
        throw new Error(
          `Template with name: ${templateOwner}/${templateRepo} not found, ${e.message}`,
        );
      }

      if (!repoTemplate?.is_template) {
        ctx.logger.warn(
          `Invalid template repository provided in templateUrl, is the repository a GitHub template?`,
        );
        throw new InputError(
          'Invalid template repository provided in templateUrl, is the repository a GitHub template?',
        );
      }
  const templateRepoCreationPromise = client.rest.repos.createUsingTemplate({
```